### PR TITLE
Added a utility of mHM2OGS

### PR DESCRIPTION
**Pull request by wenqing**
This PR
1. adds a utility of mHM2OGS (based on an old toolkit, GIS2FEM).
2. moves some members of CFEMesh to the new class, mHMPreprocessor,  for mHM2OGS.
3. removes unused codes in msh_mesh.h/cpp and removes some other unused files.

### CODE REVIEW

**Comment by wenqing** [comment]
@norihiro-w  In this year, I am preparing the last version of ogs5, ogs5.8. The final version includes what are presented in the PR, merging what developed by BGR for decovelax, improvement in the excavation modelling, bugfixing, code cleaning and so on. Your comments are very appreciated!

File: FEM/rf_pcs.cpp [hunk 1]
```diff
@@ -13635,73 +13636,64 @@ void CRFProcess::UpdateTransientBC()
...
 			double* g_para = precip->GIS_shape_head;
-			long size = (long)(g_para[0] * g_para[1]);
+			std::size_t ncols = static_cast<std::size_t>(g_para[0]);
+			std::size_t nrows = static_cast<std::size_t>(g_para[1]);
+			std::size_t size = ncols * nrows;
+			const double cell_area = g_para[4] * g_para[4];
 			vector<double> cell_data_p(size);
 			vector<double> cell_data_v(size);
-			for (i = 0; i < size; i++)
+			for (std::size_t i = 0; i < size; i++)
```
Comment (norihiro-w) [inline FEM/rf_pcs.cpp]: `i` is used in line 13620. please check if there is any problem

File: MSH/msh_mesh.cpp [hunk 2]
```diff
@@ -4583,5 +3575,102 @@ void CFEMesh::GetConnectedElements(std::vector<long>& nodes_on_sfc, std::vector<
...
+			for (int k = 0; k < 3; k++)
+				(*transform_tensor)(k, 2) *= fac;
+
+#ifdef output_top_z
+			for (k = 0; k < elem->GetNodesNumber(quad); k++)
+				node_mark[elem->GetNodeIndex(k)] = bottom;
+#endif
+		}
+		else
+			elem->SetMark(false);
```
Comment (norihiro-w) [inline MSH/msh_mesh.cpp]: are you sure if the element marks are set true at initial?

**Comment by norihiro-w** [comment]
@wenqing i have only minor comments

File: FEM/rf_pcs.cpp (hunk 1 above)
Comment (wenqing) [reply FEM/rf_pcs.cpp]: There is no problem in computation but this variable name causes confusion. I have changed the source according in the new commit.

File: MSH/msh_mesh.cpp (hunk 2 above)
Comment (wenqing) [reply MSH/msh_mesh.cpp]: Here the face elements on the top surface are marked by the condition of
n * z >0, where n is the normal vector of an face element, and z=(0, 0, 1)^T. The orientation of the normal vector is guaranteed by a vector from the the center of 3D element to an node of the face.

In this loop over face elements, elements that satisfy n * z >0 are marked as true, and the others are marked as false. Therefore there is no need to set initial status of the face elements.

**Comment by wenqing** [comment]
@norihiro-w  Thanks for your comments.

File: MSH/msh_mesh.cpp (hunk 2 above)
Comment (wenqing) [reply MSH/msh_mesh.cpp]: In the new commit, I moved the ``else if`` for the bottom surface to the section of `` #ifdef output_top_z``.

**Comment by norihiro-w** [comment]
looks good from my side

**Pull request merged**